mitmproxy/net/http/request.py

```diff
@@ -367,13 +367,64 @@ def _get_urlencoded_form(self):
                 pass
         return ()
 
+    def _sanitized_form_data_replacement(self, form_data):
```

Comment:
We should move the logic here into a function without side effects so that it's easier to test. I think it would make sense to add a second `similar_to: str=None` argument to `mitmproxy.net.http.url.encode` and then have the logic there.